docs/FAQ.md

```diff
@@ -373,3 +373,38 @@ stripping out nested `vendor` directories.
 ## How do I configure a dependency that doesn't tag its releases?
 
 Add a constraint to `Gopkg.toml` that specifies `branch: "master"` (or whichever branch you need) in the `[[constraint]]` for that dependency. `dep ensure` will determine the current revision of your dependency's master branch, and place it in `Gopkg.lock` for you. See also: [What is the difference between Gopkg.toml and Gopkg.lock?](#what-is-the-difference-between-gopkgtoml-the-manifest-and-gopkglock-the-lock)
+
+## How do I use `dep` with Docker?
+
+`dep ensure -vendor-only` creates the vendor folder from a valid `Gopkg.toml` and `Gopkg.lock` without checking for Go code.
+This is especially useful for builds inside docker utilizing cache layers.
+
+Sample dockerfile:
+
+    FROM golang:1.8 AS builder
```

Comment:
It is better to update base image to `golang:1.9-alpine`. (`alpine` since other stage use alpine and c library may cause error.)

Reply:
I would prefer that the end image use `scratch` instead.

If you need to use Go in a container you should know how to use the correct image that satisfies your dependencies.

Reply:
How to run static go binaries in Docker isn't the responsibility of the dep FAQ. The question that we want to answer is only the part about caching the results of `dep ensure`. If this is going to trigger a "static binary in a docker container" bikeshed, or if we are worried that we have to represent a perfect best practice example in our FAQ, then I vote we rip this out and only show an example up to running `dep ensure`.

Reply:
I agree. Removing the parts after `dep ensure` is a great idea.

Reply:
I agree too. Going to go ahead and do that.